website/content/platform/development/contributing/how-to/add_data_to_existing_endpoint.md

```diff
@@ -0,0 +1,860 @@
+    """Fixture to setup obb."""
+    if pytestconfig.getoption("markexpr") != "not integration":
+        import openbb
+
+        return openbb.obb
+
+@parametrize(
+    "params",
+    [
+        ({"symbol": "AAPL", "limit": 5, "provider": "fmp"}),
+        (
+            {
+                "symbol": "AAPL",
+                "period": "quarter",
+                "limit": 5,
+                "provider": "alpha_vantage"
+            }
+        ),
+    ],
+)
+@pytest.mark.integration
+def test_equity_fundamental_historical_eps(params, obb):
+    params = {p: v for p, v in params.items() if v}
+
+    result = obb.equity.fundamental.historical_eps(**params)
+    assert result
+    assert isinstance(result, OBBject)
+    assert len(result.results) > 0
+```
+
+Run this test by navigating into the folder above and entering:
+
+```console
+pytest test_equity_python.py
+```
+
+:::note
+
+If tests not related to the items being touched directly are failing, don't worry about them. That's out of scope.
+
+:::
+
+#### API Test
+
+The API test is slightly different, but the `params` can be copied and pasted from the Python test. In addition to basic checks, it will fail when values returned are not JSON serializable.
+
+```python
+"""API integration tests for equity extension."""
+
+import base64
+from datetime import time
+
+import pytest
+import requests
+from extensions.tests.conftest import parametrize
+from openbb_core.env import Env
+from openbb_core.provider.utils.helpers import get_querystring
+
+# pylint: disable=too-many-lines,redefined-outer-name
+
+
+@pytest.fixture(scope="session")
+def headers():
+    userpass = f"{Env().API_USERNAME}:{Env().API_PASSWORD}"
+    userpass_bytes = userpass.encode("ascii")
+    base64_bytes = base64.b64encode(userpass_bytes)
+
+    return {"Authorization": f"Basic {base64_bytes.decode('ascii')}"}
+
+
+@parametrize(
+    "params",
+    [
+        ({"symbol": "AAPL", "limit": 5, "provider": "fmp"}),
+        (
+            {
+                "symbol": "AAPL",
+                "period": "quarter",
+                "limit": 5,
+                "provider": "alpha_vantage"
+            }
+        ),
+    ],
+)
+@pytest.mark.integration
+def test_equity_fundamental_historical_eps(params, headers):
+    params = {p: v for p, v in params.items() if v}
+
+    query_str = get_querystring(params, [])
+    url = f"http://0.0.0.0:8000/api/v1/equity/fundamental/historical_eps?{query_str}"
+    result = requests.get(url, headers=headers, timeout=10)
+    assert isinstance(result, requests.Response)
+    assert result.status_code == 200
+```
+
+To run this test, we will need to open a second terminal and start the server.
+
+```console
+uvicorn openbb_core.api.rest_api:app
+```
+
+Go back to the other terminal and run the test by entering:
+
+```console
+pytest test_equity_api.py
+```
+
+Step 8 is done.
+  - [X] Add integration tests.
+
+
+All that's left now is to submit the work as a pull request for review.
```

Comment:
Maybe mention that they can run a test for the tests to see if they are added properly.